File: M/1561-maximum-number-of-coins-you-can-get/main.go
package main

import (
	"fmt"
	"sort"
)

func maxCoins(piles []int) int {
	sort.Ints(piles)

	sum := 0
	for len(piles) > 0 {
		sum += piles[len(piles)-2]
		piles = piles[1 : len(piles)-2]
	}
	return sum
}

func main() {
	fmt.Println()
}

File: M/1561-maximum-number-of-coins-you-can-get/main.rs
struct Solution {}

impl Solution {
  pub fn max_coins(piles: Vec<i32>) -> i32 {
    let mut piles = piles;
    piles.sort_unstable();
    piles[piles.len() / 3..]
      .iter()
      .enumerate()
      .fold(0, |sum, (idx, v)| if idx % 2 == 0 { sum + v } else { sum })
  }
}
